src/traces/scattersmith/attributes.js

```diff
@@ -0,0 +1,70 @@
+'use strict';
+
+var hovertemplateAttrs = require('../../plots/template_attributes').hovertemplateAttrs;
+var texttemplateAttrs = require('../../plots/template_attributes').texttemplateAttrs;
+var extendFlat = require('../../lib/extend').extendFlat;
+var scatterAttrs = require('../scatter/attributes');
+var lineAttrs = scatterAttrs.line;
+
+module.exports = {
+    mode: scatterAttrs.mode,
+
+    re: {
+        valType: 'data_array',
+        editType: 'calc+clearAxisTypes',
+        description: 'Sets the real coordinates'
+    },
+
+    im: {
+        valType: 'data_array',
+        editType: 'calc+clearAxisTypes',
+        description: 'Sets the imaginary coordinates'
+    },
+
+    text: scatterAttrs.text,
+    texttemplate: texttemplateAttrs({editType: 'plot'}, {
+        keys: ['re', 'im', 'text']
+    }),
+    hovertext: scatterAttrs.hovertext,
+
+    line: {
+        color: lineAttrs.color,
+        width: lineAttrs.width,
+        dash: lineAttrs.dash,
+        shape: extendFlat({}, lineAttrs.shape, {
+            values: ['linear', 'spline']
```

Comment:
@alexcjohnson wondering `spline` may be a better `dflt` here?

Reply:
I think I prefer `linear` as the default. `spline` gets closer to where intermediate points would have been in many common cases, but there are important cases where it's worse, and `linear` makes it clearer where the actual data points are. So I'd prefer a default of `linear`, with `spline` as opt-in.